src/spaces/grid_single.jl

```diff
@@ -110,3 +110,15 @@ function nearby_ids(
     a::A, model::ABM{<:GridSpaceSingle{D},A}, r = 1) where {D,A<:AbstractAgent}
     return nearby_ids(a.pos, model, r, offsets_within_radius_no_0)
 end
+
+# Method for walking on GridSpaceSingle, similar to `walk!` in spaces/utilities.jl
+function walk!(
+    agent::AbstractAgent,
+    direction::NTuple{D,Int},
+    model::ABM{<:GridSpaceSingle}
+) where {D}
+    target = normalize_position(agent.pos .+ direction, model)
+    if id_in_position(target, model) == 0 # if target unoccupied
```

Comment:
```suggestion
    if isempty(target, model) # if target unoccupied
```

please use the API function here instead of manually checking.